services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,869 @@
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = false
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created without protected path"
+        !northbound.getFlowPath(flow.id).protectedPath
+
+        when: "Update flow: enable protected path"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update flow: Not enough bandwidth found or path not found. " +
+                "Couldn't find non overlapping protected path"
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    def "Able to create a flow with protected path when there is not enough bandwidth and ignoreBandwidth=true"() {
+        given: "Two active neighboring switches"
+        def isls = topology.getIslsForActiveSwitches()
+        def (srcSwitch, dstSwitch) = [isls.first().srcSwitch, isls.first().dstSwitch]
+
+        and: "Update all ISLs which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { database.updateIslAvailableBandwidth(it, 90) }
+
+        when: "Create flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        northbound.getFlowPath(flow.id).protectedPath
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { database.resetIslBandwidth(it) }
+    }
+
+    def "System is able to recalculate protected path when protected path is broken"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def allIsls = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = getNotNeighboringSwitchPair(2)
+
+        when: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+
+        and: "Current path is not equal to protected path"
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        currentPath != currentProtectedPath
+
+        and: "Bandwidth is reserved for protected path on involved ISLs"
```

Comment:
```
and: "Current path is not equal to protected path"
and: "Bandwidth is reserved for protected path on involved ISLs"
```
I don't mind having these verifications, but do they really deserve separate documentation blocks given the test name is `System is able to recalculate protected path when protected path is broken`